nerdlets/github-about/main.js

```diff
@@ -209,10 +221,52 @@ export default class GithubAbout extends React.PureComponent {
       documentId: 'githubUrl'
     };
     await AccountStorageMutation.mutate(mutation);
+    this.setState({ accountGithubUrl: '', githubAccessError: null });
+  }
+
+  async _getGithubUrl() {
+    const { data: ghUrlObj } = await UserSecretsQuery.query({
+      name: 'GH_URL'
+    });
+    if (ghUrlObj) {
+      const { value: githubUrl } = ghUrlObj;
+      this.setState({ githubUrl });
+    }
+    if (!ghUrlObj && GITHUB_URL) {
+      this.setState({ githubUrl: GITHUB_URL });
+    }
+  }
+
+  async _setGithubUrl(githubUrl) {
+    githubUrl = formatGithubUrl(githubUrl);
+    if (githubUrl === '') {
+      this.setState({ githubUrl });
+      return;
+    }
+    const mutation = {
+      actionType: UserSecretsMutation.ACTION_TYPE.WRITE_SECRET,
+      name: 'GH_URL',
+      value: githubUrl
+    };
+    const { data } = await UserSecretsMutation.mutate(mutation);
+    const status = get(data, 'nerdStorageVaultWriteSecret.status');
+    if (status === 'SUCCESS') {
+      this.setState({ githubUrl });
+    }
+    return status;
+  }
+
+  async _deleteGithubUrl() {
+    const mutation = {
+      actionType: UserSecretsMutation.ACTION_TYPE.DELETE_SECRET,
+      name: 'GH_URL'
+    };
+    await UserSecretsMutation.mutate(mutation);
     this.setState({ githubUrl: '', githubAccessError: null });
```

Comment:
Do we need to check for any errors with the `DELETE_SECRET` and `WRITE_SECRET` operations? I could see us needing to check (and potentially set) `githubAccessError` in the event that we get that response back from the API.